src/dialog/dialog.ts

```diff
@@ -0,0 +1,40 @@
+import {Injectable, ComponentRef, DynamicComponentLoader, Injector, Component} from 'angular2/core';
+
+const CONTAINER_TAG_NAME = 'ngb-dialog-container';
+
+@Component({selector: CONTAINER_TAG_NAME, template: ''})
+class NgbDialogContainer {
+}
+
+export class NgbDialogOpenResult {
+  constructor(public componentRefs: ComponentRef[]) {}
+
+  disposeAll() {
+    this.componentRefs.forEach((component) => { component.dispose(); });
```

Comment:
`NgbDialogOpenResult` itself shouldn't be (too) long-living, so not sure it matters. Also I don't know how "expensive" `ComponentRef` is...

In short: I don't know, but adding it wouldn't hurt.